### ISSUE TITLE
update to egui 0.20.0

### ISSUE THREAD

**Issue by ctaggart** [tauri-apps/tauri-egui#18]
egui 0.20.0 was released two weeks ago.
https://github.com/emilk/egui/releases/tag/0.20.0

I would really like to use the new version with tauri-egui.

**Comment by ctaggart** [comment]
I attempted to apply the tao changes on top of equi-0.20.1. I got egui-winit to compile, but not eframe.

https://github.com/ctaggart/egui/pull/2

This was based on the changes from this repo, which you can see in this PR:

https://github.com/ctaggart/egui/pull/1

**Comment by ctaggart** [comment]
@wusyong, excited to see recent work on tauri-egui to use the Tauri v2 alphas! Any chance the egui version can be bumped too? https://crates.io/crates/egui-tao/0.22.0/dependencies is still stuck on https://crates.io/crates/egui/0.19.0 . The current version is now https://crates.io/crates/egui/0.22.0 .

**Comment by denjell-crabnebula** [comment]
There was work going on here by @dklassic :

https://github.com/tauri-apps/egui/pull/19

but gtk3 dowsn't have vulcan context :(

**Comment by ctaggart** [comment]
Looks like https://github.com/tauri-apps/egui/pull/19 just needs to be reviewed and merged.

**Comment by ctaggart** [comment]
We can see that https://crates.io/crates/eframe_tao/0.23.0/dependencies published  on June 15th and has a dependency on egui ^0.22.0.

**Comment by dklassic** [comment]
Actually, don't close this just yet.
tauri-egui did indeed need some rework to be able to use egui 0.22 and I'm still working on it!

**Comment by ctaggart** [comment]
@dklassic, is this still in progress?

**Comment by dklassic** [comment]
> @dklassic, is this still in progress?

@wusyong took over the development a while ago and I think it is actually complete for a while now: https://github.com/tauri-apps/tauri-egui/tree/0.22

@wusyong: is there a reason for why you haven't opened a PR for tauri-egui 0.22?

**Comment by wusyong** [comment]
Oops sorry for the late reply. I would like to wait for tauri v2 published beta version.
Because right now alpha version can lead to breaking change too easily which make it even harder to maintain the plugin.
And by the time v2 release, it probably already have another egui version.

**Comment by ctaggart** [comment]
Thanks @wusyong. I'm really looking forward to this. I see the v2 roadmap was published on September 7th.
https://beta.tauri.app/blog/roadmap-to-tauri-2-0/

**Comment by ShayBox** [comment]
This probably shouldn't have updated to Tauri v2 and just stuck to v1 since it's going to be at-least another year or two until v2 reaches beta, rc, or stable. This project is essentially abandoned until then, unless someone capable wants to make their own or continue it with a fork. I wish I hadn't decided to use this for my project because now it's stuck on an early version of Tauri v2 alpha and egui 0.22 for the forseable future.

**Comment by FabianLars** [comment]
> This probably shouldn't have updated to Tauri v2 and just stuck to v1

Agreed, but i think there was actually a somewhat good reason for it but can't remember exactly so maybe that's wishful thinking.

> since it's going to be at-least another year or two until v2 reaches beta, rc, or stable

Who told you this? Beta is still planned for Q1 and literally right around the corner. Tauri is only missing one major change (large part of the work already done) and the audit of the inner libs (wry, tao) already begun...

> unless someone capable wants to make their own or continue it with a fork

Hmm, i would hope that just because we don't have resources for it ourselves doesn't mean we couldn't handle pull requests. But then again, with beta almost being here i i think it's worth to wait a bit longer.

> now it's stuck on an early version of Tauri v2 alpha and egui 0.22 for the forseable future.

(Ignoring the tauri version) Out of pure interest as a non-egui user, what do you need/want from the newer egui versions?

**Comment by ctaggart** [comment]
Thanks for the update @FabianLars! I’m excited to see this land this quarter.

**Comment by ctaggart** [comment]
Tauri v2 Beta released on February 2nd  🚀  https://beta.tauri.app/blog/tauri-2-0-0-beta/

A lot of new features and fixes have landed between egui 0.20.0 and the latest 0.26.2.
https://github.com/emilk/egui/releases

I am really looking forward to the new integration. Is there still a possibility of an early beta landing this month?

**Comment by dklassic** [comment]
My current plan is to wait for the release of egui 0.27 and I'll try to catch it as fast as I can. Not sure if I'll do a 0.26 or 0.27 by then, we'll see.

I'll try to move more swiftly this time around.

**Comment by patchedsoul** [comment]
> My current plan is to wait for the release of egui 0.27 and I'll try to catch it as fast as I can. Not sure if I'll do a 0.26 or 0.27 by then, we'll see.
> 
> I'll try to move more swiftly this time around.

Any updates now that egui 0.27.x is out?

**Comment by FabianLars** [comment]
@dklassic is currently looking into it but it doesn't look a trivial change.

**Comment by ctaggart** [comment]
Awesome to see the [2.0 Release Candidate](https://v2.tauri.app/blog/tauri-2-0-0-release-candidate/)! @dklassic, any luck with this or it too big a change?

**Comment by dklassic** [comment]
> Awesome to see the [2.0 Release Candidate](https://v2.tauri.app/blog/tauri-2-0-0-release-candidate/)! @dklassic, any luck with this or it too big a change?

Thanks for tagging me, I've been working on porting on and off a while ago and it's probably best to share some thoughts about current state of porting `egui`.

The biggest issue to porting stems down to the now drastically different structuring of `tao` and `winit`. The porting process is basically figuring out the difference and handle it as best as I can, but `egui` 0.27 itself is quite different from how it works back in 0.22 and so is `winit` since then. So not only the porting is a bit painful and dragging, I'm not particularly confident in porting all the subsequent version changes in a timely manner.

I think I'll attempt another push at it but we might need more help if we were to continue the `tauri-egui` support.

**Comment by clearlysid** [comment]
Hi folks! Some good news → I've gotten a basic version of this setup working again [here](https://github.com/clearlysid/egui-tao) along with some thoughts on how we can perhaps offer a way forward in the README.

Happy to accelerate work on this if there's some community interest. But in the meantime, I will keep proceeding slowly atleast until my own use-cases for a `tauri-egui` setup are solved.

Cheers!
